dll/win32/msgina/shutdown.c

```diff
@@ -510,20 +550,21 @@ ShutdownDialog(
     Context.ShutdownOptions = ShutdownOptions;
     Context.bCloseDlg = FALSE;
     Context.bReasonUI = GetShutdownReasonUI();
+    Context.bLunaUI = GetShutdownLunaUI();
 
-    if (pgContext->hWlx && pgContext->pWlxFuncs)
+    if (pgContext->hWlx && pgContext->pWlxFuncs && !Context.bLunaUI)
     {
         ret = pgContext->pWlxFuncs->WlxDialogBoxParam(pgContext->hWlx,
                                                       pgContext->hDllInstance,
-                                                      MAKEINTRESOURCEW(Context.bReasonUI ? IDD_SHUTDOWN_REASON : IDD_SHUTDOWN),
+                                                      MAKEINTRESOURCEW(Context.bLunaUI ? IDD_SHUTDOWN_FANCY : (Context.bReasonUI ? IDD_SHUTDOWN_REASON : IDD_SHUTDOWN)),
                                                       hwndDlg,
                                                       ShutdownDialogProc,
                                                       (LPARAM)&Context);
     }
     else
     {
         ret = DialogBoxParamW(pgContext->hDllInstance,
-                              MAKEINTRESOURCEW(Context.bReasonUI ? IDD_SHUTDOWN_REASON : IDD_SHUTDOWN),
+                              MAKEINTRESOURCEW(Context.bLunaUI ? IDD_SHUTDOWN_FANCY : (Context.bReasonUI ? IDD_SHUTDOWN_REASON : IDD_SHUTDOWN)),
```

Comment:
Double ternary, I would say, is a sign that you should introduce a variable for the dialog id.